rules/no-array-for-each.js

```diff
@@ -175,13 +180,35 @@ function getFixFunction(callExpression, functionInfo, context) {
 			return false;
 		}
 
-		if (callback.body.type !== 'BlockStatement') {
+		if (callback.body.type !== 'BlockStatement' && !isOptionalChaining) {
 			return false;
 		}
 
 		return true;
 	};
 
+	function * wrapInIfStatement(fixer) {
+		const isSingleLine = !isBlockStatement || callback.body.loc.start.line === callback.body.loc.end.line;
+
+		yield fixer.insertTextBefore(callExpression, `if (${callExpression.callee.object.name}) {\n`);
```

Comment:
This is not right, `callExpression.callee.object` not always `Indentifier`, `foo.bar()?.forEach(...)`

Reply:
Good catch! There's an even simpler example: `foo()?.forEach(...)`. The question is how do we handle such cases? I guess we can either auto fix it to:
```
				if (foo()) {
					for (const element of foo()) {
						...
					}
				}
```
which might cause issues if `foo` produces side effects. Alternatively, we can fix it so that `foo()` is stored in a variable or we could NOT fix such cases. WDYT?

Reply:
Use `suggest` for cases has side effect.

Reply:
How can we know whether there's a side effect? Also, what would be the suggested fix? My above snippet or you're thinking about more auto-fixed suggestions?

Reply:
[`hasSideEffect()`](https://eslint-utils.mysticatea.dev/api/ast-utils.html#hassideeffect).


Store `foo()` in a variable should be the best (and can use auto-fix instead of suggestion), but I'm afraid it's hard to implement.

We can suggest your code above, unless you want do more.

Reply:
I think I've finished implementing suggestion